### PR TITLE
USA and env-specific  banners

### LINKED ISSUE #3579: Add "official site of US government" banner
### Summary

**What we're after:**
As a website user, I want to be confident the site I'm looking at is legitimately a US government website. 

We should add a "official site of US government" banner to fec.gov. This is recommended by the US Web Design Standards and also requested by @wluoFEC.

**Background:**
From [US Web Design System website](https://designsystem.digital.gov/components/header/)

>We recommend that all federal government sites include the “official government website” banner and a logo or site name. The banner clearly identifies your site as part of the United States government, while a logo or site name helps visitors understand who you are and what you do

**Code:** https://github.com/uswds/uswds/blob/b8730e8cd17ac451343699e73039c1a7f9a50ebb/src/components/banner/banner.njk

**Example:**
https://www.gsa.gov
Screen Shot 2020-02-28 at 2 34 09 PM

 **Completion criteria**
- [ ] Add a "official site of US government" banner to fec.gov based on [US Web Design Standards](https://designsystem.digital.gov/components/header/)

### LINKED ISSUE #3678: Add development / staging environment banners
### Summary

**What we're after:**
_We need to have a banner on our feature, development, and stage environments so that users know that they are NOT viewing the live production website._

This used to be on our website but has disappeared. See example screenshot below. 

Screen Shot 2020-04-09 at 10 46 13 AM

### Completion criteria

- Ensure that the environment banners are on each of these environments:
   - [ ] Feature
   - [ ] Dev
   - [ ] Stage
- [ ] Ensure that this banner does not appear on the production website

### PR DESCRIPTION
**Pull request by johnnyporkchops**
## Summary
- Add Official Gov Website banner 
- Add environment-specific banners with conditional statement in base templates 
- Remove redundant scss imports by refactoring  elections.scss so that is does not import global scss  twice
  -  remove @import `base.scss` 
  - move everything in `base.scss` that's not in `global.scss` into `elections.scss`\
  -  remove stylesheet reference to  the compiled `base.css` in `home_base.html`

**Note:**
Hardcoding the three `img src` paths in the `use-banner.html`  allows us to use one partial for the whole site, rather than having one for Django and one for Jinja. Hardcoding was a way to bypass syntax differences between Django/Jinja for the `static path variable` in templates.

- Resolves #3579 
- Resolves #3678 

## Impacted areas of the application

modified:   data/templates/layouts/main.jinja
modified:   fec/settings/base.py
modified:   fec/static/scss/components/_site-header.scss
modified:   fec/static/scss/elections.scss
modified:   fec/templates/base.html
modified:   fec/templates/home_base.html
new file:    fec/templates/partials/usa-banner.html
new file:    data/templates/partials/data-usa-banner.jinja

## Screenshots
Apr-18-2020 01-07-19
Moile: 
Apr-19-2020 22-11-36

## Related PRs
https://github.com/fecgov/fec-cms/pull/3611
https://github.com/fecgov/fec-cms/pull/3610

## How to test
 - checkout `feature/3579-3678-usa-and-env-banners`
- `npm run build-sass`
-  `runserver` and review  local env-specific banner and usa-gov banner
- See https://fec-feature-cms.app.cloud.gov to confirm  Feature env-specific banner

____

### CODE REVIEW

File: fec/fec/templates/partials/env-banner.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+	  {% if settings.FEC_CMS_ENVIRONMENT != 'PRODUCTION' %}   
+	      <div class="banner ">
+	        <p class="t-sans t-small">{{settings.FEC_CMS_ENVIRONMENT}}<br>
+	        This site is for testing ideas and code. View the most accurate data on the <a href="http://www.fec.gov">live site</a>.</p>
+	      </div>
+	    {% endif %}
+     {% else %}  
+     {% if FEC_CMS_ENVIRONMENT != 'PRODUCTION' %}  
+      <div class="banner ">
+        <p class="t-sans t-small">{{FEC_CMS_ENVIRONMENT}}X<br>
```
Comment (patphongs) [inline fec/fec/templates/partials/env-banner.html]: Looks like an extra `X` is being inserted after the environment name here

File: fec/fec/templates/partials/env-banner.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{% if settings  != indefined %}
```
Comment (patphongs) [inline fec/fec/templates/partials/env-banner.html:1]: Typo? Should be `undefined`

File: fec/fec/templates/partials/env-banner.html (hunk 1 above)
Comment (patphongs) [reply fec/fec/templates/partials/env-banner.html]: This was fixed already

**Comment by patphongs** [comment]
> I tried changing the FEC_CMS_ENVIRONMENT variable locally to dev, stage, and prod and it did not render on the homepage. But it was ok on some other pages. I can't tell if this would be an issue when we go live.

After reviewing this further with @johnnyporkchops, it appears this happens locally when running our development branch as well. This shouldn't be an issue on our cloud environments.

**Comment by JonellaCulmer** [comment]
@johnnyporkchops Also, does the USWDS guidance for this banner specifically mention placement on the left? Up until now we've placed it on the right.

**Comment by lbeaufort** [comment]
> @johnnyporkchops Also, does the USWDS guidance for this banner specifically mention placement on the left? Up until now we've placed it on the right.

@JonellaCulmer I don't think so (docs: https://designsystem.digital.gov/components/banner/) but my thinking was that on the left it appears closer to the URL, which is what is being verified.

**Comment by johnnyporkchops** [comment]
@JonellaCulmer --
Screen Shot 2020-04-21 at 2 31 09 AM

**Comment by johnnyporkchops** [comment]
> > @johnnyporkchops Also, does the USWDS guidance for this banner specifically mention placement on the left? Up until now we've placed it on the right.
> 
> @JonellaCulmer I don't think so (docs: https://designsystem.digital.gov/components/banner/) but my thinking was that on the left it appears closer to the URL, which is what is being verified. In addition, it's most commonly on the left on other US government sites (and by default in the USWDS example) so it would be good to be consistent with what is most commonly seen on other US gov sites.

I was going to show left, center, and right screenshots as it does seem like it might be nice to have it on the left but assumed we would end up opting to follow USWDS and other gov site implementations.  If we are still considering this, I can do a screenshot

File: tasks.py [hunk 3]
```diff
@@ -75,7 +75,7 @@ def _detect_space(repo, branch=None, yes=False):
     ('stage', lambda _, branch: branch.startswith('release')),
     ('dev', lambda _, branch: branch == 'develop'),
     # Uncomment below and adjust branch name to deploy desired feature branch to the feature space
-    #('feature', lambda _, branch: branch == '[BRANCH NAME]'),
+    ('feature', lambda _, branch: branch == 'feature/3579-3678-usa-and-env-banners'),
```
Comment (lbeaufort) [inline tasks.py:78]: Can you remove this deploy rule change before we merge?

File: fec/fec/templates/partials/env-banner.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+	      </div>
+	    {% endif %}
+     {% else %}  
+     {% if FEC_CMS_ENVIRONMENT != 'PRODUCTION' %}  
+      <div class="banner ">
+        <p class="t-sans t-small">{{FEC_CMS_ENVIRONMENT}}<br>
+        This site is for testing ideas and code. View the most accurate data on the <a href="http://www.fec.gov">live site</a>.</p>
+      </div>
+      {% endif %}
+     {% endif %}
```
Comment (lbeaufort) [inline fec/fec/templates/partials/env-banner.html:15]: Can you align the `if/endif` for readability please?

File: fec/fec/templates/base.html [hunk 5]
```diff
@@ -87,9 +87,6 @@ <h2 style="color: #ffffff">Your browser is outdated</h2>
     {% else %}
     <header class="site-header">
       <div class="masthead">
```
Comment (lbeaufort) [inline fec/fec/templates/base.html:89]: Will leaving this `header` and adding the partial for the banner in the next file give us two headers? Will investigate too.

**Comment by JonellaCulmer** [comment]
@johnnyporkchops Thanks for that. I'm fine using the same pattern as other government sites and leaving it on the left. Particularly if that's where other citizens are looking for that when on a government site. I was just curious about their guidance. 

No need to spend time with different views.

File: fec/fec/templates/base.html (hunk 5 above)
Comment (johnnyporkchops) [reply fec/fec/templates/base.html:89]: Yes. I considered the same issue before doing this-- I researched html5 specs that say it's not illegal to have more than one `header` tag per document.

Comment (johnnyporkchops) [reply fec/fec/templates/base.html:89]: `header` specs:
https://html.spec.whatwg.org/multipage/sections.html#the-header-element

**Comment by lbeaufort** [comment]
I also tested in IE and it looks great 👍

**Comment by johnnyporkchops** [comment]
I have addressed and made changes for all review comments. I think this is ready, unless there are add'l comments or something I missed.

**Comment by lbeaufort** [comment]
@johnnyporkchops great! I changed the target branch to be the release. Sorry I just noticed - is it hard to make the **https** bold like in the USWDS docs? https://designsystem.digital.gov/components/banner/

**Pull request merged**